### PR TITLE
feat: created tiplink for Sol and spl tokens

### LINKED ISSUE #21: support Tiplink API
https://docs.tiplink.io/

### PR DESCRIPTION
**Pull request by shivaji43**
# Pull Request Description

## Related Issue
Fixes #21

## Changes Made
This PR adds the following changes:

- Added support for tiplinks for SOL 
- Also supports spl tokens

## Transaction executed by agent 

Example transaction: 
Screenshot 2024-12-26 115547
Screenshot 2024-12-26 115606

## Prompt Used

```create me a tiplink for 0.001 sol```

## Checklist
- [x] I have tested these changes locally
- [ ] I have updated the documentation
- [x] I have added a transaction link
- [x] I have added the prompt used to test it

### CODE REVIEW

**Comment by shivaji43** [comment]
@thearyanag you can now check the code , I've tested it locally

File: src/langchain/index.ts [hunk 1]
```diff
@@ -1229,6 +1229,67 @@ export class SolanaCreateGibworkTask extends Tool {
   }
 }
 
+export class SolanaTipLinkTool extends Tool {
+  name = "solana_tiplink";
+  description = `Create a TipLink for transferring SOL or SPL tokens.
```
Comment (thearyanag) [inline src/langchain/index.ts:1234]: can you make the description similar to rest tools, you can take the transfer tool for reference

Comment (shivaji43) [reply src/langchain/index.ts:1234]: yes i will do it immediately

Comment (shivaji43) [reply src/langchain/index.ts:1234]: @thearyanag done

**Comment by shivaji43** [comment]
@thearyanag thank you for being patient with me through this

**Pull request merged**